File: apps/mybookkeeper/frontend/src/shared/types/tax/source-document.ts
export interface TaxSourceDocument {
  document_id: string;
  file_name: string | null;
  document_type: string;
  issuer_name: string | null;
  issuer_ein: string | null;
  tax_year: number;
  key_amount: number | null;
  source: string;
  uploaded_at: string;
  form_instance_id: string;
}

export interface ChecklistItem {
  expected_type: string;
  expected_from: string | null;
  reason: string;
  status: "received" | "missing";
  document_id: string | null;
}

export interface SourceDocumentsResponse {
  documents: TaxSourceDocument[];
  checklist: ChecklistItem[];
}

File: apps/mybookkeeper/backend/app/schemas/tax/source_document.py
"""Schemas for the source documents endpoint."""
import uuid
from datetime import datetime

from pydantic import BaseModel


class TaxSourceDocument(BaseModel):
    document_id: uuid.UUID
    file_name: str | None
    document_type: str
    issuer_name: str | None
    issuer_ein: str | None
    tax_year: int
    key_amount: float | None
    source: str
    uploaded_at: datetime
    form_instance_id: uuid.UUID


class ChecklistItem(BaseModel):
    expected_type: str
    expected_from: str | None
    reason: str
    status: str  # "received" | "missing"
    document_id: uuid.UUID | None = None


class SourceDocumentsResponse(BaseModel):
    documents: list[TaxSourceDocument]
    checklist: list[ChecklistItem]
